### ISSUE TITLE
[Accessibility]: VoiceOver reads CodeMirror placeholder text as if it's editable

### ISSUE THREAD

**Issue by limitedmage** [sourcegraph/sourcegraph#43135]
### Audit type

Screen reader navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/34188

### Problem description

VoiceOver reads the code mirror placeholder as if it's editable, even saying where the "cursor" is in relation to the placeholder.

<img width="699" alt="image" src="https://user-images.githubusercontent.com/206864/196556447-3e16f9b8-f7bf-4239-bf41-62dff21a5491.png">

### Expected behavior

The placeholder should not be read as if it's editable.

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [ ] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [ ] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.

/cc @lguychard

**Comment by fkling** [comment]
How was this tested? When I use voice over with Safari 16 it seems to work (no audio):

https://user-images.githubusercontent.com/179026/196638087-e60ea303-c4ee-41f3-857b-9b824e0d10aa.mov

The element that contains the placeholder text has already `contenteditable="false"` set, so I'm not quite sure what's going on here.

**Comment by fkling** [comment]
So, it reads the placeholder value for me but doesn't seem to treat it as editable text.

**Comment by limitedmage** [comment]
Yes, I did all testing on Chrome for this. I'd be fine with either the placeholder text not being read at all (as long as the field is correctly labeled and the screen reader reads the label on focus). Would making the placeholder `aria-hidden` work here? Or are deeper, codemirror-specific changes needed?

**Comment by fkling** [comment]
We are using CodeMirror's extension for that but if I remember correctly it's a rather simple one so we could just duplicate the code and make the appropriate changes.

**Comment by limitedmage** [comment]
Sounds good, let's give that a shot. If we can resolve this, it might be worth submitting a patch to the official extension with the fix.

**Comment by limitedmage** [comment]
Looking into this with more detail, instead/in addition to making the placeholder `aria-hidden`, we should [add an `aria-placeholder`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-placeholder#:~:text=The%20aria%2Dplaceholder%20attribute%20defines,description%20of%20the%20expected%20format.) to the search box with the placeholder text.
